Frends.Regex.MatchResult/Frends.Regex.MatchResult/Definitions/Result.cs

```diff
@@ -0,0 +1,18 @@
+﻿namespace Frends.Regex.MatchResult.Definitions;
+using System.Text.RegularExpressions;
+
+/// <summary>
+/// Result.
+/// </summary>
+public class Result
+{
+    internal Result(MatchCollection matches)
+    {
+        Matches = matches;
+    }
+
+    /// <summary>
+    /// Match result.
+    /// </summary>
+    public MatchCollection Matches { get; private set; }
```

Comment:
For compatibility and future we do not use builtin .NET types for returning anything. This needs a wrapper class and be like `public RegexMatch[] Matches {...}` or something similar. Refer to https://learn.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.match?view=net-9.0#properties for properties that we can have in our wrapper class. At least Index, GroupName and Value should be in the output.